### PR TITLE
Us313138 jwt signing

### PR DESCRIPTION
**Pull request by witrisna**

### CODE REVIEW

File: mag/src/main/java/com/ca/mas/core/context/MssoContext.java [hunk 1]
```diff
@@ -319,7 +318,7 @@ protected void onConnectionObtained(HttpURLConnection connection) {
      * <p/>
      * Setting a non-null ID token has the side-effect of causing any cached password in memory to be destroyed.
      *
-     * @param idToken the ID token.  Required.
+     * @param idToken the ID token.  Required.Ø
```
Comment (ForceKinetics) [inline mag/src/main/java/com/ca/mas/core/context/MssoContext.java]: Was this intended?

File: mas/src/main/java/com/ca/mas/foundation/JwtSignRequest.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+                JWSSigner signer = new RSASSASigner(getPrivateKey());
+                JWTClaimsSet.Builder claimBuilder = new JWTClaimsSet.Builder();
+                TokenManager tokenManager = StorageProvider.getInstance().getTokenManager();
+
+                // JWT claims
+                // iss
+                String magId = tokenManager.getMagIdentifier();
+                claimBuilder.issuer("device://" + magId + "/$");
+
+                // TODO: sub: username
```
Comment (ForceKinetics) [inline mas/src/main/java/com/ca/mas/foundation/JwtSignRequest.java]: Are we merging this with no solution for this field?

File: mag/src/main/java/com/ca/mas/core/context/MssoContext.java (hunk 1 above)
Comment (witrisna) [inline mag/src/main/java/com/ca/mas/core/context/MssoContext.java]: > No, it was not intended. The sign is removed.

File: mas/src/main/java/com/ca/mas/foundation/JwtSignRequest.java (hunk 2 above)
Comment (witrisna) [inline mas/src/main/java/com/ca/mas/foundation/JwtSignRequest.java]: The JwtSignRequest has been removed, it has been replaced with com.ca.mas.foundation.MASRequestBody#jwtClaimsBody, com.ca.mas.foundation.MAS#sign(com.ca.mas.foundation.MASClaims) and 
com.ca.mas.foundation.MAS#sign(com.ca.mas.foundation.MASClaims, java.security.PrivateKey)

**Pull request merged**